### ISSUE TITLE
pre-commit hook run only on staged/committed files on current branch?

### ISSUE THREAD

**Issue by natarajanc-prodigygame** [jendrikseipp/vulture#259]
vulture seems like a very useful tool, yet I can't configure it quite right. Perhaps it's a bug - or a configuration issue that could be explained slightly better.

i want to use the vulture pre-commit hook to only run on the staged/committed files in my branch individually and not run it based on directories specified in `paths` under the pyproject.toml

If i use the `pyproject.toml` config and specify the directories - the unused imported lib in 1 file can be used in another file in the same directory and hence the detection of unused imports does not happen

is it possible to only run on staged/committed files individually?

**Comment by RJ722** [comment]
If Vulture is run on staged/committed files only, it *will* report false "unused import" results.

I'm not sure how pre-commit works under the hood, but I'm wondering if it can be configured to do the following: Run Vulture on all your files, and then grep the results to show only the ones for staged/committed files.

**Comment by natarajanc-prodigygame** [comment]
@RJ722  taking a simple example - assuming 2 files in same directory

file a.py
 import lib_a  # unused in this file

file b.py
 import lib_a   # actually used  in the file 


if i run vulture on all files , the unused import for file `a.py` wont be surfaced (even with grep solution you indicated)

the unused import for file a.py only happens when i run `vulture a.py` which is what i want

**Comment by anita-clement** [comment]
Hi, is there an update on this issue? 

I'm running into the same issue. `vulture` is running on all files when I'm running pre-commit while the expected behaviour from pre-commit is for it to run on staged files only. Thanks

**Comment by jendrikseipp** [comment]
I agree with @RJ722 : it's important to run Vulture on all files. Otherwise, if you edit a file containing name definitions but these names are not used in the file, you'll get many false positives.

As @RJ722 says: if it's possible to only **report** the unused code from changed files, that would be a useful change, though. PRs welcome :-)
